src/Attributes/HookCustom.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Drush\Attributes;
+
+use Attribute;
+use Consolidation\AnnotatedCommand\Parser\CommandInfo;
+
+#[Attribute(Attribute::TARGET_METHOD)]
+class HookCustom
+{
+    /**
+     * @param $name
+     *  The hook name which is being implemented.
+     * @param $arguments
+     *   Arguments which should be passed along to the hook implementation.
+     */
+    public function __construct(
+        public string $name,
+        public array $arguments = [],
+    ) {
+    }
+
+    public static function handle(\ReflectionAttribute $attribute, CommandInfo $commandInfo)
+    {
+        $instance = $attribute->newInstance();
+        $commandInfo->addAnnotation($instance->name, implode(' ', $instance->arguments));
```

Comment:
The old annotation implementation had no choice but to use an untyped top-level annotation as its selector. Now, though, we have a typed attribute, a Hook Selector, so we could and probably should be more discerning. We could add a list of selectors to the commandInfo, and set data there instead of directly in the annotations.

Alternately, we could go the other way, and instead name this `CLI\Annotation`, and make it a general-purpose way to set generic annotations on a CLI attribute. We currently don't have any other use for this facility than a hook selector, but we might someday.

Reply:
These are optional suggestions. The first is only possible in the annotations library itself, as we would have to also change the way hook selection worked. That's probably not a good idea, as it would prevent interoperation between the attribute implementation and the annotation implementation of commands. That may or may not be an issue, but with commands and hooks being defined in multiple modules, it likely would become an issue.

I don't know that the second suggestion is good either; I'm more inclined towards the specific name, unless we specifically think someone might misuse it to set generic annotation values. If we did, then maybe the more generic name might be better after all.

Reply:
I like the first proposal here. Lets do it at some future date. IMO annotations should be deprecated by this library and removed at some far future date.